tests/test_blackd.py

```diff
@@ -209,3 +209,13 @@ async def test_cors_headers_present(self) -> None:
         response = await self.client.post("/", headers={"Origin": "*"})
         self.assertIsNotNone(response.headers.get("Access-Control-Allow-Origin"))
         self.assertIsNotNone(response.headers.get("Access-Control-Expose-Headers"))
+
+    @unittest_run_loop
+    async def test_preserve_newline(self) -> None:
+        for data in (b"c\r\nc\r\n", b"l\nl\n"):
```

Comment:
Could we also test:
- first is only n, second is rn, and the corresponding expectation
- the other way around

Reply:
I'm not sure what you mean, could you provide an example please.

Reply:
Yep exactly, sorry for the confusion 😅 At a quick glance, I couldn't find a test with mixed line endings. Only ones like the one you wrote [here](https://github.com/psf/black/blob/main/tests/test_black.py#L1264). But it would be nice, and a short mention on documentation couldn't hurt either, agreed 👍 Just to be explicit.

Reply:
I have made a sus discovery, black will not normalize newlines if the rest of the file is unchanged. Should I make blackd behave exactly the same? (I think always normalizing is arguably better).

Reply:
I added tests for both blackd and black, and updated the docs.